src/modules.js

```diff
@@ -1,17 +1,39 @@
+import axios from "axios";
+import Bowser from "bowser";
+import firebase from "firebase/app";
+import 'firebase/auth';
 /* global __analytics */
 // Note there has been special combination of if statement applied here so the unused code can be eliminated
 // during dead code elimination process, while editing/refactoring this file please take care,
 // so that the elimination process still works.
 // Also note that each analytics cannot be split into separate files, as it will break code elimination.
 const analyticModules = [];
-
+const filterProperties = (allowProperties = [], disallowProperties = [], properties) => {
+  let eventProperties = properties;
+  if(allowProperties.length > 0 ) {
+    eventProperties =  Object.keys(eventProperties)
+      .reduce((object, key) => (allowProperties.includes(key)
+          ? {...object, [key]: eventProperties[key]}
+          : object
+      ), {});
+  }
+  if(disallowProperties.length > 0 ) {
+    eventProperties =  Object.keys(eventProperties)
+      .reduce((object, key) => (disallowProperties.includes(key)
+          ? object
+          : {...object, [key]: eventProperties[key]}
+      ), {});
+  }
+  return eventProperties;
```

Comment:
H: The function has potential to reduce the computational complexity to only one reduce/loop (as right now there many be meny object repacking). It can be done by first calculating a list of keys to copy/remove based on `allowProperties`, `disallowProperties` and `Object.keys(eventProperties)`.

Reply:
thanks!